### DISCUSSION TITLE
How are the source2sdk json files generated?

### DISCUSSION THREAD

**Discussion by roflmuffin** [saul/demofile-net#2] [Q&A] [answered]
I am heading up a project that allows you to write C# plugin scripts for CS2, and I am leveraging your json files & sdk gen project to generate C# classes & structs for use with the schema system: https://github.com/roflmuffin/CounterStrikeSharp

However there are some fields in the server.hpp/json file for instance that are marked as `NetworkDisable` so don't generate in your json file, however they are present on the server and accessible via schema system, so I would like to include these in the generated output.

Is the json file generator that you have modified source available? if so that would be great so I could modify & generate the json files as well. Thanks :pray:

**Comment by saul** [comment]
Hi - yes I plan to release a fork of source2gen soon - it's on the todo list :) 

Note that CounterStrikeSharp is licensed under GPL, but both source2gen and this project are MIT licensed. I see that you've copied [DemoFile.SdkGen](https://github.com/saul/demofile-net/tree/main/src/DemoFile.SdkGen) under your project without attribution. MIT projects are required to have attribution in substantial copies of the code - see for example the [ACKNOWLEDGEMENTS](https://github.com/saul/demofile-net/blob/main/ACKNOWLEDGEMENTS) file in this repo.

**Reply by roflmuffin** [reply]
Thanks I have included an acknowledgements with an MIT license and reference, there is also a small preamble in the Program.cs file for the schema generator. Let me know if more is required.

**Comment by saul** [comment, marked as answer]
The code for CS2SchemaGen lives here: https://github.com/saul/CS2SchemaGen

Hope that helps
